src/charm.py

```diff
@@ -339,44 +357,35 @@ def _render_conf_file(self) -> str:
             template = Template(file.read())
 
         default_schema_id, schemas = self._get_identity_schema_config()
+        oidc_providers = self._get_oidc_providers()
         login_ui_url = self._get_login_ui_endpoint_info("login_url")
+        mappers = self._get_claims_mappers()
         rendered = template.render(
             cookie_secrets=[self._get_secret()],
             log_level=self._log_level,
-            mappers_path=str(self._mappers_dir_path),
+            mappers=mappers,
             default_browser_return_url=self._get_login_ui_endpoint_info("login_url"),
             allowed_return_urls=[login_ui_url] if login_ui_url else [],
             identity_schemas=schemas,
             default_identity_schema_id=default_schema_id,
-            public_base_url=self._public_url,
             login_ui_url=login_ui_url,
             error_ui_url=self._get_login_ui_endpoint_info("error_url"),
-            oidc_providers=self.external_provider.get_providers(),
+            oidc_providers=oidc_providers,
             available_mappers=self._get_available_mappers,
-            db_info=self._get_database_relation_info(),
             oauth2_provider_url=self._get_hydra_endpoint_info(),
             smtp_connection_uri=self.config.get("smtp_connection_uri"),
         )
         return rendered
 
-    def _push_file(
-        self, dst: Path, src: Optional[Path] = None, content: Optional[str] = None
-    ) -> None:
-        if not content:
-            if not src:
-                raise ValueError("`src` or `content` must be provided.")
-            with open(src) as f:
-                content = f.read()
-        self._container.push(dst, content, make_dirs=True)
-
-    def _push_default_files(self) -> None:
-        for schema_file in self._mappers_local_dir_path.iterdir():
-            self._push_file(self._mappers_dir_path / schema_file.name, src=schema_file)
-
-        for schema_file in self._identity_schemas_local_dir_path.iterdir():
-            self._push_file(
-                self._identity_schemas_default_dir_path / schema_file.name, src=schema_file
-            )
+    @retry(
```

Comment:
Have a question: is it possible that we can remove the retry mechanism since we move the `dsn` to env vars?

Reply:
The reason for the retry is that if someone (admin UI) else tries to patch the configmap at the same time, we will get an error (because we will have an older revision). If this happens, we need to either catch the error and defer the event or retry. 

Since it is quite possible that our request gets a transient error (as described above), I think that it makes sense to retry instead of failing right away.